ros2/diagnostics/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+cmake_minimum_required(VERSION 3.8)
+project(diagnostics)
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(diagnostic_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
+
+set(dependencies
+rclcpp
+diagnostic_msgs
+std_msgs
+)
```

Comment:
Any particular reason for adding this variable here? It's only used once, so better just list these dependencies in `ament_target_dependencies`

Reply:
It's just a habit. I usually do this when I have several nodes with the same dependencies in the same package. It's true that for one single executable it maybe not worth having it this way.